lib/rspec/rails/matchers/have_reported_error.rb

```diff
@@ -0,0 +1,195 @@
+              return "Expected error message to match #{@expected_error}, but got: '#{actual_error.message}'"
+            when Symbol
+              return "Expected error to be #{@expected_error}, but got: #{actual_error}"
+            else
+              return "Expected specific error, but got #{actual_error.class} with message: '#{actual_error.message}'"
+            end
+          end
+        end
+
+        def failure_message_when_negated
+          error_count = @error_subscriber.events.count
+          error_word = 'error'.pluralize(error_count)
+          verb = error_count == 1 ? 'has' : 'have'
+
+          "Expected the block not to report any errors, but #{error_count} #{error_word} #{verb} been reported."
+        end
+
+        private
+
+        def error_matches_expectation?
+          return true if @expected_error.nil? && @error_subscriber.events.any?
+
+          case @expected_error
+          when Class
+            actual_error.is_a?(@expected_error)
+          when Exception
+            actual_error.is_a?(@expected_error.class) &&
+              (@expected_error.message.empty? || actual_error.message == @expected_error.message)
+          when Regexp
+            actual_error.message&.match(@expected_error)
+          when Symbol
+            actual_error == @expected_error
+          end
+        end
+
+        def attributes_match_if_specified?
+          return true if @attributes.empty?
+          return false if @error_subscriber.events.empty?
+
+          event_context = @error_subscriber.events.last.attributes[:context]
+          attributes_match?(event_context)
+        end
+
+        def actual_error
+          @error_subscriber.events.empty? ? nil : @error_subscriber.events.last.error
+        end
+
+        def attributes_match?(actual)
+          @attributes.all? do |key, value|
+            if value.respond_to?(:matches?)
+              value.matches?(actual[key])
+            else
+              actual[key] == value
+            end
+          end
+        end
+
+        def unmatched_attributes(actual)
+          @attributes.reject do |key, value|
+            if value.respond_to?(:matches?)
+              value.matches?(actual[key])
+            else
+              actual[key] == value
+            end
+          end
+        end
+      end
+
+      # @api public
+      # Passes if the block reports an error to `Rails.error`.
+      #
+      # This matcher asserts that ActiveSupport::ErrorReporter has received an error report.
+      #
+      # @example Checking for any error
+      #   expect { Rails.error.report(StandardError.new) }.to have_reported_error
+      #
+      # @example Checking for specific error class
+      #   expect { Rails.error.report(MyError.new) }.to have_reported_error(MyError)
+      #
+      # @example Checking for specific error instance with message
+      #   expect { Rails.error.report(MyError.new("message")) }.to have_reported_error(MyError.new("message"))
```

Comment:
I think this should be broken into either two arguments e.g.
```
have_reported_error(MyError, "message")
have_reported_error(MyError, /me/)
```
Or instead use `with(message:)`